src/helpers/judgeZero.ts

```diff
@@ -0,0 +1,119 @@
+import axios from 'axios';
+import Eris, { Embed, EmbedOptions, MessageContent } from 'eris';
+
+const client = axios.create({
+    baseURL: 'https://judge0-ce.p.rapidapi.com',
+    headers: {
+        'content-type': 'application/json',
+        'x-rapidapi-key': process.env.JUDGE0_API_KEY,
+        'x-rapidapi-host': 'judge0-ce.p.rapidapi.com',
+        useQueryString: true,
+    },
+});
+
+interface JudgeZeroLanguage{
+    id : number,
+    name : string
+}
+
+interface JudgeZeroRequestBody{
+    language_id : number
+    source_code : string
+}
+
+interface JudgeZeroPostResponse{
+    token : string
+}
+
+interface JudgeZeroGetResponse{
+    stdout : string
+    stderr : string
+    message : string
+    memory : number
+    time : string
+    compile_output : string
+}
+
+const judgeZeroSubmissionParams = {
+    base64_encoded: true,
+    fields: '*',
+};
+
+export async function getAvailableLanguages() : Promise<JudgeZeroLanguage[]> {
+    const res = await client.get('/languages');
+    return res.data;
+}
+
+export async function createEvalRequest(body : JudgeZeroRequestBody) : Promise<JudgeZeroPostResponse> {
+    body.source_code = Buffer.from(body.source_code, 'binary').toString('base64');
+
+    const res = await client.post('/submissions', body, {
+        params: judgeZeroSubmissionParams,
+    });
+
+    return res.data;
+}
+
+export async function getEvalResult(token : string) : Promise<JudgeZeroGetResponse> {
+    const res = await client.get(`/submissions/${token}`, {
+        params: judgeZeroSubmissionParams,
+    });
+
+    const { data } = res;
+
+    data.stdout = Buffer.from(data.stdout, 'base64').toString('binary');
+
+    if (data.compile_output != null || '') {
+        data.compile_output = Buffer.from(data.compile_output, 'base64').toString('binary');
+    }
+
+    return data;
+}
+
+export function genereateJudgeZeroMessage(originalMsg : Eris.Message<Eris.TextableChannel>, result : JudgeZeroGetResponse) : MessageContent {
+    if (result.stderr) {
+        let msg = `An error occured when evaluating your code: **${result.stderr}**`;
+        if (result.message) {
+            msg += `\nHere's some extra message for you: **${result.message}**`;
+        }
+        return { content: msg };
+    }
+
+    if (!result.stdout) {
+        if (result.compile_output) {
+            const msg = `Sorry! I cannot compile your code. Here's some error message: ${result.compile_output}`;
+            return { content: msg };
```

Comment:
Also do note that the compile output *can* be over 2048 characters if it has stack trace.